src/math/Math.sol

```diff
@@ -33,9 +33,8 @@ library Math {
         }
     }
 
-    /// @dev Returns the floor of log2(x) and returns 0 on input 0.
-    /// @dev This computation makes use of De Bruijn sequences, their usage for computing log2 is referenced here: http://supertech.csail.mit.edu/papers/debruijn.pdf
-    function log2(uint256 x) internal pure returns (uint256 y) {
+    /// @dev Returns the biggest power of 2 smaller than or equal to x and returns 0 on input 0.
+    function _roundDownToPowerOf2(uint256 x) internal pure returns (uint256) {
```

Comment:
```suggestion
    function _log2Down(uint256 x) internal pure returns (uint256) {
```

(this is a suggestion to make it less verbose)

Reply:
I don't really like it because it seems like we are applying log2 and then rounding down (but instead we find the power of 2 that is immediatly less than the input)

Reply:
Agree with @QGarchery. Otherwise `_onlyHighestSetBit` works too, but not especially better.